// oceanic.js/dist/lib/util/Routes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CHANNEL_JOINED_PRIVATE_ARCHIVED_THREADS = exports.CHANNEL_INVITES = exports.CHANNEL_FOLLOWERS = exports.CHANNEL_BULK_DELETE_MESSAGES = exports.CHANNEL = exports.GUILD_INCIDENT_ACTIONS = exports.GUILD_ONBOARDING = exports.GUILD_STICKERS = exports.GUILD_STICKER = exports.GUILD_VOICE_STATE = exports.GUILD_WELCOME_SCREEN = exports.GUILD_WIDGET_JSON = exports.GUILD_WIDGET_IMAGE = exports.GUILD_VANITY_URL = exports.GUILD_WIDGET = exports.GUILD_INTEGRATIONS = exports.GUILD_INTEGRATION = exports.GUILD_INVITES = exports.GUILD_PRUNE = exports.GUILD_MFA = exports.GUILD_ROLE = exports.GUILD_ROLES = exports.GUILD_MEMBER_ROLE = exports.GUILD_SEARCH_MEMBERS = exports.GUILD_MEMBER = exports.GUILD_MEMBERS = exports.GUILD_ACTIVE_THREADS = exports.GUILD_CHANNELS = exports.GUILD_PREVIEW = exports.GUILD_TEMPLATES = exports.GUILD_TEMPLATE = exports.GUILD_TEMPLATE_CODE = exports.GUILD_SCHEDULED_EVENT_USERS = exports.GUILD_SCHEDULED_EVENTS = exports.GUILD_SCHEDULED_EVENT = exports.GUILD_AUDIT_LOG = exports.GUILD_EMOJIS = exports.GUILD_EMOJI = exports.GUILD_AUTOMOD_RULES = exports.GUILD_AUTOMOD_RULE = exports.GUILD_BAN = exports.GUILD_BANS = exports.GUILDS = exports.GUILD = exports.WEBHOOK_PLATFORM = exports.WEBHOOK_MESSAGE = exports.WEBHOOK = exports.GUILD_WEBHOOKS = exports.CHANNEL_WEBHOOKS = exports.CDN_URL = void 0;
exports.TEAM_ICON = exports.STICKER_PACK_BANNER = exports.STICKER = exports.ROLE_ICON = exports.MEMBER_BANNER = exports.GUILD_SPLASH = exports.GUILD_SCHEDULED_EVENT_COVER = exports.GUILD_ICON = exports.GUILD_DISCOVERY_SPLASH = exports.GUILD_AVATAR = exports.EMBED_AVATAR = exports.CUSTOM_EMOJI = exports.BANNER = exports.APPLICATION_ICON = exports.APPLICATION_COVER = exports.APPLICATION_ASSET = exports.ACHIEVEMENT_ICON = exports.OAUTH_ROLE_CONNECTION = exports.OAUTH_TOKEN_REVOKE = exports.OAUTH_TOKEN = exports.OAUTH_GUILDS = exports.OAUTH_GUILD_MEMBER = exports.OAUTH_GUILD = exports.OAUTH_CONNECTIONS = exports.OAUTH_CHANNELS = exports.OAUTH_CURRENT_USER = exports.OAUTH_INFO = exports.OAUTH_AUTHORIZE = exports.OAUTH_APPLICATION = exports.VOICE_STATUS = exports.GUILD_VOICE_REGIONS = exports.VOICE_REGIONS = exports.GROUP_RECIPIENT = exports.CHANNEL_TYPING = exports.CHANNEL_THREAD_MEMBERS = exports.CHANNEL_THREAD_MEMBER = exports.CHANNEL_THREADS = exports.CHANNEL_REACTION_USER = exports.CHANNEL_REACTIONS = exports.CHANNEL_REACTION = exports.CHANNEL_PUBLIC_ARCHIVED_THREADS = exports.CHANNEL_PRIVATE_ARCHIVED_THREADS = exports.CHANNEL_PINS = exports.CHANNEL_PINNED_MESSAGE = exports.CHANNEL_PERMISSIONS = exports.CHANNEL_PERMISSION = exports.CHANNEL_MESSAGE_THREADS = exports.CHANNEL_MESSAGES_CROSSPOST = exports.CHANNEL_MESSAGES = exports.CHANNEL_MESSAGE = void 0;
exports.APPLICATION = exports.ROLE_CONNECTIONS_METADATA = exports.STAGE_INSTANCE = exports.STAGE_INSTANCES = exports.INVITE = exports.NITRO_STICKER_PATHS = exports.STICKER_PACKS = exports.MESSAGE_LINK = exports.USER = exports.GATEWAY_BOT = exports.GATEWAY = exports.INTERACTION_CALLBACK = exports.GUILD_APPLICATION_COMMAND_PERMISSIONS = exports.GUILD_APPLICATION_COMMAND_PERMISSION = exports.GUILD_APPLICATION_COMMANDS = exports.GUILD_APPLICATION_COMMAND = exports.APPLICATION_COMMANDS = exports.APPLICATION_COMMAND = exports.USER_AVATAR_DECORATION = exports.USER_AVATAR = void 0;
/** @module Routes */
/* eslint-disable @typescript-eslint/explicit-function-return-type */
exports.CDN_URL = "https://cdn.discordapp.com";
// Webhooks
const CHANNEL_WEBHOOKS = (channelID) => `/channels/${channelID}/webhooks`;
exports.CHANNEL_WEBHOOKS = CHANNEL_WEBHOOKS;
const GUILD_WEBHOOKS = (guildID) => `/guilds/${guildID}/webhooks`;
exports.GUILD_WEBHOOKS = GUILD_WEBHOOKS;
const WEBHOOK = (webhookID, webhookToken) => (webhookToken ? `/webhooks/${webhookID}/${webhookToken}` : `/webhooks/${webhookID}`);
exports.WEBHOOK = WEBHOOK;
const WEBHOOK_MESSAGE = (webhookID, webhookToken, messageID) => `/webhooks/${webhookID}/${webhookToken}/messages/${messageID}`;
exports.WEBHOOK_MESSAGE = WEBHOOK_MESSAGE;
const WEBHOOK_PLATFORM = (webhookID, webhookToken, platform) => `/webhooks/${webhookID}/${webhookToken}/${platform}`;
exports.WEBHOOK_PLATFORM = WEBHOOK_PLATFORM;
// Guilds
const GUILD = (userID) => `/guilds/${userID}`;
exports.GUILD = GUILD;
exports.GUILDS = "/guilds";
const GUILD_BANS = (guildID) => `/guilds/${guildID}/bans`;
exports.GUILD_BANS = GUILD_BANS;
const GUILD_BAN = (guildID, userID) => `/guilds/${guildID}/bans/${userID}`;
exports.GUILD_BAN = GUILD_BAN;
const GUILD_AUTOMOD_RULE = (guildID, autoModerationRuleID) => `/guilds/${guildID}/auto-moderation/rules/${autoModerationRuleID}`;
exports.GUILD_AUTOMOD_RULE = GUILD_AUTOMOD_RULE;
const GUILD_AUTOMOD_RULES = (guildID) => `/guilds/${guildID}/auto-moderation/rules`;
exports.GUILD_AUTOMOD_RULES = GUILD_AUTOMOD_RULES;
const GUILD_EMOJI = (guildID, emojiID) => `/guilds/${guildID}/emojis/${emojiID}`;
exports.GUILD_EMOJI = GUILD_EMOJI;
const GUILD_EMOJIS = (guildID) => `/guilds/${guildID}/emojis`;
exports.GUILD_EMOJIS = GUILD_EMOJIS;
const GUILD_AUDIT_LOG = (guildID) => `/guilds/${guildID}/audit-logs`;
exports.GUILD_AUDIT_LOG = GUILD_AUDIT_LOG;
const GUILD_SCHEDULED_EVENT = (guildID, eventID) => `/guilds/${guildID}/scheduled-events/${eventID}`;
exports.GUILD_SCHEDULED_EVENT = GUILD_SCHEDULED_EVENT;
const GUILD_SCHEDULED_EVENTS = (guildID) => `/guilds/${guildID}/scheduled-events`;
exports.GUILD_SCHEDULED_EVENTS = GUILD_SCHEDULED_EVENTS;
const GUILD_SCHEDULED_EVENT_USERS = (guildID, eventID) => `/guilds/${guildID}/scheduled-events/${eventID}/users`;
exports.GUILD_SCHEDULED_EVENT_USERS = GUILD_SCHEDULED_EVENT_USERS;
const GUILD_TEMPLATE_CODE = (code) => `/guilds/templates/${code}`;
exports.GUILD_TEMPLATE_CODE = GUILD_TEMPLATE_CODE;
const GUILD_TEMPLATE = (guildID, templateID) => `/guilds/${guildID}/templates/${templateID}`;
exports.GUILD_TEMPLATE = GUILD_TEMPLATE;
const GUILD_TEMPLATES = (guildID) => `/guilds/${guildID}/templates`;
exports.GUILD_TEMPLATES = GUILD_TEMPLATES;
const GUILD_PREVIEW = (guildID) => `/guilds/${guildID}/preview`;
exports.GUILD_PREVIEW = GUILD_PREVIEW;
const GUILD_CHANNELS = (guildID) => `/guilds/${guildID}/channels`;
exports.GUILD_CHANNELS = GUILD_CHANNELS;
const GUILD_ACTIVE_THREADS = (guildID) => `/guilds/${guildID}/threads/active`;
exports.GUILD_ACTIVE_THREADS = GUILD_ACTIVE_THREADS;
const GUILD_MEMBERS = (guildID) => `/guilds/${guildID}/members`;
exports.GUILD_MEMBERS = GUILD_MEMBERS;
const GUILD_MEMBER = (guildID, userID) => `/guilds/${guildID}/members/${userID}`;
exports.GUILD_MEMBER = GUILD_MEMBER;
const GUILD_SEARCH_MEMBERS = (guildID) => `/guilds/${guildID}/members/search`;
exports.GUILD_SEARCH_MEMBERS = GUILD_SEARCH_MEMBERS;
const GUILD_MEMBER_ROLE = (guildID, userID, roleID) => `/guilds/${guildID}/members/${userID}/roles/${roleID}`;
exports.GUILD_MEMBER_ROLE = GUILD_MEMBER_ROLE;
const GUILD_ROLES = (guildID) => `/guilds/${guildID}/roles`;
exports.GUILD_ROLES = GUILD_ROLES;
const GUILD_ROLE = (guildID, roleID) => `/guilds/${guildID}/roles/${roleID}`;
exports.GUILD_ROLE = GUILD_ROLE;
const GUILD_MFA = (guildID) => `/guilds/${guildID}/mfa`;
exports.GUILD_MFA = GUILD_MFA;
const GUILD_PRUNE = (guildID) => `/guilds/${guildID}/prune`;
exports.GUILD_PRUNE = GUILD_PRUNE;
const GUILD_INVITES = (guildID) => `/guilds/${guildID}/invites`;
exports.GUILD_INVITES = GUILD_INVITES;
const GUILD_INTEGRATION = (guildID, integrationID) => `/guilds/${guildID}/integrations/${integrationID}`;
exports.GUILD_INTEGRATION = GUILD_INTEGRATION;
const GUILD_INTEGRATIONS = (guildID) => `/guilds/${guildID}/integrations`;
exports.GUILD_INTEGRATIONS = GUILD_INTEGRATIONS;
const GUILD_WIDGET = (guildID) => `/guilds/${guildID}/widget`;
exports.GUILD_WIDGET = GUILD_WIDGET;
const GUILD_VANITY_URL = (guildID) => `/guilds/${guildID}/vanity-url`;
exports.GUILD_VANITY_URL = GUILD_VANITY_URL;
const GUILD_WIDGET_IMAGE = (guildID) => `/guilds/${guildID}/widget.png`;
exports.GUILD_WIDGET_IMAGE = GUILD_WIDGET_IMAGE;
const GUILD_WIDGET_JSON = (guildID) => `/guilds/${guildID}/widget.json`;
exports.GUILD_WIDGET_JSON = GUILD_WIDGET_JSON;
const GUILD_WELCOME_SCREEN = (guildID) => `/guilds/${guildID}/welcome-screen`;
exports.GUILD_WELCOME_SCREEN = GUILD_WELCOME_SCREEN;
const GUILD_VOICE_STATE = (guildID, userID) => `/guilds/${guildID}/voice-states/${userID}`;
exports.GUILD_VOICE_STATE = GUILD_VOICE_STATE;
const GUILD_STICKER = (guildID, stickerID) => `/guilds/${guildID}/stickers/${stickerID}`;
exports.GUILD_STICKER = GUILD_STICKER;
const GUILD_STICKERS = (guildID) => `/guilds/${guildID}/stickers`;
exports.GUILD_STICKERS = GUILD_STICKERS;
const GUILD_ONBOARDING = (guildID) => `/guilds/${guildID}/onboarding`;
exports.GUILD_ONBOARDING = GUILD_ONBOARDING;
const GUILD_INCIDENT_ACTIONS = (guildID) => `/guilds/${guildID}/incident-actions`;
exports.GUILD_INCIDENT_ACTIONS = GUILD_INCIDENT_ACTIONS;
// Channels
const CHANNEL = (channelID) => `/channels/${channelID}`;
exports.CHANNEL = CHANNEL;
const CHANNEL_BULK_DELETE_MESSAGES = (channelID) => `/channels/${channelID}/messages/bulk-delete`;
exports.CHANNEL_BULK_DELETE_MESSAGES = CHANNEL_BULK_DELETE_MESSAGES;
const CHANNEL_FOLLOWERS = (channelID) => `/channels/${channelID}/followers`;
exports.CHANNEL_FOLLOWERS = CHANNEL_FOLLOWERS;
const CHANNEL_INVITES = (channelID) => `/channels/${channelID}/invites`;
exports.CHANNEL_INVITES = CHANNEL_INVITES;
const CHANNEL_JOINED_PRIVATE_ARCHIVED_THREADS = (channelID) => `/channels/${channelID}/users/@me/threads/archived/private`;
exports.CHANNEL_JOINED_PRIVATE_ARCHIVED_THREADS = CHANNEL_JOINED_PRIVATE_ARCHIVED_THREADS;
const CHANNEL_MESSAGE = (channelID, messageID) => `/channels/${channelID}/messages/${messageID}`;
exports.CHANNEL_MESSAGE = CHANNEL_MESSAGE;
const CHANNEL_MESSAGES = (channelID) => `/channels/${channelID}/messages`;
exports.CHANNEL_MESSAGES = CHANNEL_MESSAGES;
const CHANNEL_MESSAGES_CROSSPOST = (channelID, messageID) => `/channels/${channelID}/messages/${messageID}/crosspost`;
exports.CHANNEL_MESSAGES_CROSSPOST = CHANNEL_MESSAGES_CROSSPOST;
const CHANNEL_MESSAGE_THREADS = (channelID, messageID) => `/channels/${channelID}/messages/${messageID}/threads`;
exports.CHANNEL_MESSAGE_THREADS = CHANNEL_MESSAGE_THREADS;
const CHANNEL_PERMISSION = (channelID, overwriteID) => `/channels/${channelID}/permissions/${overwriteID}`;
exports.CHANNEL_PERMISSION = CHANNEL_PERMISSION;
const CHANNEL_PERMISSIONS = (channelID) => `/channels/${channelID}/permissions`;
exports.CHANNEL_PERMISSIONS = CHANNEL_PERMISSIONS;
const CHANNEL_PINNED_MESSAGE = (channelID, messageID) => `/channels/${channelID}/pins/${messageID}`;
exports.CHANNEL_PINNED_MESSAGE = CHANNEL_PINNED_MESSAGE;
const CHANNEL_PINS = (channelID) => `/channels/${channelID}/pins`;
exports.CHANNEL_PINS = CHANNEL_PINS;
const CHANNEL_PRIVATE_ARCHIVED_THREADS = (channelID) => `/channels/${channelID}/threads/archived/private`;
exports.CHANNEL_PRIVATE_ARCHIVED_THREADS = CHANNEL_PRIVATE_ARCHIVED_THREADS;
const CHANNEL_PUBLIC_ARCHIVED_THREADS = (channelID) => `/channels/${channelID}/threads/archived/public`;
exports.CHANNEL_PUBLIC_ARCHIVED_THREADS = CHANNEL_PUBLIC_ARCHIVED_THREADS;
const CHANNEL_REACTION = (channelID, messageID, reaction) => `/channels/${channelID}/messages/${messageID}/reactions/${reaction}`;
exports.CHANNEL_REACTION = CHANNEL_REACTION;
const CHANNEL_REACTIONS = (channelID, messageID) => `/channels/${channelID}/messages/${messageID}/reactions`;
exports.CHANNEL_REACTIONS = CHANNEL_REACTIONS;
const CHANNEL_REACTION_USER = (channelID, messageID, reaction, user) => `/channels/${channelID}/messages/${messageID}/reactions/${reaction}/${user}`;
exports.CHANNEL_REACTION_USER = CHANNEL_REACTION_USER;
const CHANNEL_THREADS = (channelID) => `/channels/${channelID}/threads`;
exports.CHANNEL_THREADS = CHANNEL_THREADS;
const CHANNEL_THREAD_MEMBER = (channelID, userID) => `/channels/${channelID}/thread-members/${userID}`;
exports.CHANNEL_THREAD_MEMBER = CHANNEL_THREAD_MEMBER;
const CHANNEL_THREAD_MEMBERS = (channelID) => `/channels/${channelID}/thread-members`;
exports.CHANNEL_THREAD_MEMBERS = CHANNEL_THREAD_MEMBERS;
const CHANNEL_TYPING = (channelID) => `/channels/${channelID}/typing`;
exports.CHANNEL_TYPING = CHANNEL_TYPING;
const GROUP_RECIPIENT = (channelID, userID) => `/channels/${channelID}/recipients/${userID}`;
exports.GROUP_RECIPIENT = GROUP_RECIPIENT;
exports.VOICE_REGIONS = "/voice/regions";
const GUILD_VOICE_REGIONS = (guildID) => `/guilds/${guildID}/regions`;
exports.GUILD_VOICE_REGIONS = GUILD_VOICE_REGIONS;
const VOICE_STATUS = (channelID) => `/channels/${channelID}/voice-status`;
exports.VOICE_STATUS = VOICE_STATUS;
// OAuth
exports.OAUTH_APPLICATION = "/oauth2/applications/@me";
exports.OAUTH_AUTHORIZE = "/oauth2/authorize";
exports.OAUTH_INFO = "/oauth2/@me";
exports.OAUTH_CURRENT_USER = "/users/@me";
exports.OAUTH_CHANNELS = "/users/@me/channels";
exports.OAUTH_CONNECTIONS = "/users/@me/connections";
const OAUTH_GUILD = (guildID) => `/users/@me/guilds/${guildID}`;
exports.OAUTH_GUILD = OAUTH_GUILD;
const OAUTH_GUILD_MEMBER = (guildID) => `${(0, exports.OAUTH_GUILD)(guildID)}/member`;
exports.OAUTH_GUILD_MEMBER = OAUTH_GUILD_MEMBER;
exports.OAUTH_GUILDS = "/users/@me/guilds";
exports.OAUTH_TOKEN = "/oauth2/token";
exports.OAUTH_TOKEN_REVOKE = "/oauth2/token/revoke";
const OAUTH_ROLE_CONNECTION = (applicationID) => `/users/@me/applications/${applicationID}/role-connection`;
exports.OAUTH_ROLE_CONNECTION = OAUTH_ROLE_CONNECTION;
// Images
const ACHIEVEMENT_ICON = (applicationID, achievementID, hash) => `/app-assets/${applicationID}/achievements/${achievementID}/icons/${hash}`;
exports.ACHIEVEMENT_ICON = ACHIEVEMENT_ICON;
const APPLICATION_ASSET = (applicationID, assetID) => `/applications/${applicationID}/assets/${assetID}`;
exports.APPLICATION_ASSET = APPLICATION_ASSET;
const APPLICATION_COVER = (applicationID, hash) => `/app-icons/${applicationID}/${hash}`;
exports.APPLICATION_COVER = APPLICATION_COVER;
exports.APPLICATION_ICON = exports.APPLICATION_COVER;
const BANNER = (guildOrUserID, hash) => `/banners/${guildOrUserID}/${hash}`;
exports.BANNER = BANNER;
const CUSTOM_EMOJI = (emojiID) => `/emojis/${emojiID}`;
exports.CUSTOM_EMOJI = CUSTOM_EMOJI;
const EMBED_AVATAR = (mod) => `/embed/avatars/${mod}`;
exports.EMBED_AVATAR = EMBED_AVATAR;
const GUILD_AVATAR = (guildID, userID, hash) => `/guilds/${guildID}/users/${userID}/avatars/${hash}`;
exports.GUILD_AVATAR = GUILD_AVATAR;
const GUILD_DISCOVERY_SPLASH = (guildID, hash) => `/guilds/${guildID}/splashes/${hash}`;
exports.GUILD_DISCOVERY_SPLASH = GUILD_DISCOVERY_SPLASH;
const GUILD_ICON = (guildID, hash) => `/icons/${guildID}/${hash}`;
exports.GUILD_ICON = GUILD_ICON;
const GUILD_SCHEDULED_EVENT_COVER = (eventID, hash) => `/guild-events/${eventID}/${hash}`;
exports.GUILD_SCHEDULED_EVENT_COVER = GUILD_SCHEDULED_EVENT_COVER;
const GUILD_SPLASH = (guildID, hash) => `/splashes/${guildID}/${hash}`;
exports.GUILD_SPLASH = GUILD_SPLASH;
const MEMBER_BANNER = (guildID, userID, hash) => `/guilds/${guildID}/users/${userID}/banners/${hash}`;
exports.MEMBER_BANNER = MEMBER_BANNER;
const ROLE_ICON = (roleID, hash) => `/role-icons/${roleID}/${hash}`;
exports.ROLE_ICON = ROLE_ICON;
const STICKER = (stickerID) => `/stickers/${stickerID}`;
exports.STICKER = STICKER;
const STICKER_PACK_BANNER = (assetID) => (0, exports.APPLICATION_ASSET)("710982414301790216", assetID);
exports.STICKER_PACK_BANNER = STICKER_PACK_BANNER;
const TEAM_ICON = (teamID, hash) => `/team-icons/${teamID}/${hash}`;
exports.TEAM_ICON = TEAM_ICON;
const USER_AVATAR = (userID, hash) => `/avatars/${userID}/${hash}`;
exports.USER_AVATAR = USER_AVATAR;
const USER_AVATAR_DECORATION = (userID, hash) => `/avatar-decorations/${userID}/${hash}`;
exports.USER_AVATAR_DECORATION = USER_AVATAR_DECORATION;
// Application Commands
const APPLICATION_COMMAND = (applicationID, commandID) => `/applications/${applicationID}/commands/${commandID}`;
exports.APPLICATION_COMMAND = APPLICATION_COMMAND;
const APPLICATION_COMMANDS = (applicationID) => `/applications/${applicationID}/commands`;
exports.APPLICATION_COMMANDS = APPLICATION_COMMANDS;
const GUILD_APPLICATION_COMMAND = (applicationID, guildID, commandID) => `/applications/${applicationID}/guilds/${guildID}/commands/${commandID}`;
exports.GUILD_APPLICATION_COMMAND = GUILD_APPLICATION_COMMAND;
const GUILD_APPLICATION_COMMANDS = (applicationID, guildID) => `/applications/${applicationID}/guilds/${guildID}/commands`;
exports.GUILD_APPLICATION_COMMANDS = GUILD_APPLICATION_COMMANDS;
const GUILD_APPLICATION_COMMAND_PERMISSION = (applicationID, guildID, commandID) => `/applications/${applicationID}/guilds/${guildID}/commands/${commandID}/permissions`;
exports.GUILD_APPLICATION_COMMAND_PERMISSION = GUILD_APPLICATION_COMMAND_PERMISSION;
const GUILD_APPLICATION_COMMAND_PERMISSIONS = (applicationID, guildID) => `/applications/${applicationID}/guilds/${guildID}commands/permissions`;
exports.GUILD_APPLICATION_COMMAND_PERMISSIONS = GUILD_APPLICATION_COMMAND_PERMISSIONS;
const INTERACTION_CALLBACK = (interactionID, interactionToken) => `/interactions/${interactionID}/${interactionToken}/callback`;
exports.INTERACTION_CALLBACK = INTERACTION_CALLBACK;
// Misc
exports.GATEWAY = "/gateway";
exports.GATEWAY_BOT = "/gateway/bot";
const USER = (userID) => `/users/${userID}`;
exports.USER = USER;
const MESSAGE_LINK = (guildID, channelID, messageID) => `/channels/${guildID}/${channelID}/${messageID}`;
exports.MESSAGE_LINK = MESSAGE_LINK;
exports.STICKER_PACKS = "/sticker-packs";
/** @deprecated Use {@link Routes~STICKER_PACKS | STICKER_PACKS}. This will be removed in `1.9.0`. */
exports.NITRO_STICKER_PATHS = exports.STICKER_PACKS;
const INVITE = (code) => `/invites/${code}`;
exports.INVITE = INVITE;
exports.STAGE_INSTANCES = "/stage-instances";
const STAGE_INSTANCE = (channelID) => `/stage-instances/${channelID}`;
exports.STAGE_INSTANCE = STAGE_INSTANCE;
const ROLE_CONNECTIONS_METADATA = (applicationID) => `/applications/${applicationID}/role-connections/metadata`;
exports.ROLE_CONNECTIONS_METADATA = ROLE_CONNECTIONS_METADATA;
exports.APPLICATION = "/applications/@me";
